x-pack/platform/plugins/shared/streams/server/routes/content_packs/route.ts

```diff
@@ -0,0 +1,138 @@
+    },
+  },
+  async handler({ params, request, response, getScopedClients, context }) {
+    const { assetClient, soClient, streamsClient } = await getScopedClients({ request });
+
+    await streamsClient.ensureStream(params.path.name);
+
+    const dashboards = await assetClient
+      .getAssets({ entityId: params.path.name, entityType: 'stream' })
+      .then((assets) => assets.filter(({ assetType }) => assetType === 'dashboard'));
+    if (dashboards.length === 0) {
+      throw new StatusError(`No dashboards are linked to [${params.path.name}] stream`, 400);
+    }
+
+    const exporter = (await context.core).savedObjects.getExporter(soClient);
+    const exportStream = await exporter.exportByObjects({
+      request,
+      objects: dashboards.map((dashboard) => ({ id: dashboard.assetId, type: 'dashboard' })),
+      includeReferencesDeep: true,
+    });
+
+    const savedObjects: string[] = await createPromiseFromStreams([
+      exportStream,
+      createMapStream((savedObject) => {
+        return JSON.stringify(savedObject);
+      }),
+      createConcatStream([]),
+    ]);
+
+    return response.ok({
+      body: savedObjects.join('\n'),
+      headers: {
+        'Content-Disposition': `attachment; filename="content_pack.ndjson"`,
+        'Content-Type': 'application/ndjson',
+      },
+    });
+  },
+});
+
+const uploadContentPacksRoute = createServerRoute({
+  endpoint: 'POST /api/streams/{name}/content_packs',
+  options: {
+    access: 'internal',
+    body: {
+      accepts: 'multipart/form-data',
+      output: 'stream',
+    },
+  },
+  params: z.object({
+    path: z.object({
+      name: z.string(),
+    }),
+    body: z.object({
+      content_pack: z.instanceof(Readable),
+    }),
+  }),
+  security: {
+    authz: {
+      enabled: false,
+      reason:
+        'This API delegates security to the currently logged in user and their Elasticsearch permissions.',
+    },
+  },
+  async handler({ params, request, getScopedClients, context }) {
+    const { assetClient, soClient, streamsClient } = await getScopedClients({ request });
+
+    await streamsClient.ensureStream(params.path.name);
+
+    const updatedSavedObjectsStream = await createPromiseFromStreams([
+      await createSavedObjectsStreamFromNdJson(params.body.content_pack),
+      createConcatStream([]),
+    ]);
+
+    const importer = (await context.core).savedObjects.getImporter(soClient);
+    const { successResults, errors } = await importer.import({
+      readStream: createListStream(updatedSavedObjectsStream),
+      createNewCopies: true,
+      overwrite: true,
+    });
+
+    const createdAssets = (successResults ?? [])
+      .filter((savedObject) => savedObject.type === 'dashboard')
+      .map((dashboard) => ({
+        assetType: 'dashboard' as const,
+        assetId: dashboard.destinationId ?? dashboard.id,
+      }));
+
+    if (createdAssets.length > 0) {
+      await assetClient.bulk(
```

Comment:
is this the part that adds the association between the newly created assets and the target stream?

Reply:
Correct